enhancements/network/allow-from-router-networkpolicy.md

```diff
@@ -0,0 +1,253 @@
+work with any other network plugin.
+
+In theory, with other plugins, if you knew the pod network CIDR, you
+could create a policy saying "allow from anywhere (that can route to
+the pod network) except the pod network", which is essentially
+equivalent to "allow from `default`" in OpenShift SDN. eg:
+
+    ipBlock:
+      cidr: 0.0.0.0/0
+      except:
+        - 10.128.0.0/14
+
+This is not ideal, because it requires you to know the pod network
+CIDR, but even given that, it would not actually work with many
+network plugins anyway. Eg, in ovn-kubernetes, when a node or a
+host-network pod sends traffic to a pod-network pod, the source IP is
+not the node's "primary" node IP but rather the IP associated with the
+`ovn-k8s-mp0` interface, which is inside the pod network CIDR (eg, it
+would be `10.128.6.2` on the node that has the `10.128.6.0/23`
+subnet).
+
+It would be possible to create a policy allowing specifically the
+relevant source node IP of each node, but such a policy would need to
+be updated any time nodes were added or removed from the cluster.
+
+(There has been some discussion upstream about allowing [`nodeSelector`
+in NetworkPolicies] but this is not going to happen any time soon, if
+it happens at all.)
+
+[`nodeSelector` in NetworkPolicies]: https://github.com/kubernetes/kubernetes/issues/51891
+
+### Plan
+
+There does not seem to be any good way to solve this completely
+network-plugin-agnostically at the current time, so we won't worry
+about Kuryr, Calico, etc, at this time. (Though Kuryr could choose to
+implement the same hack I propose for OVN Kubernetes.)
+
+We also don't want to diverge too much from upstream/"stock"
+NetworkPolicy, so trying to add `nodeSelector` is out.
+
+The simplest approach would be to formalize the OpenShift SDN
+host-network / `default` behavior, and add a similar behavior to OVN
+Kubernetes.
+
+The proposed OVN Kubernetes change is:
+
+  - We add a config option `host-network-namespace` to the
+    `[kubernetes]` section of the config. If this option is specified,
+    and the indicated namespace exists, then ovn-kubernetes will add
+    the relevant node IPs (management port IP and "primary" node IP)
+    of all nodes to the address set for that namespace, causing
+    node-to-pod and pod-to-node traffic to be affected by
+    NetworkPolicies that match that namespace.
+
+    (FIXME: would that actually work? Might need further tweaks...)
+
+Given that change to upstream ovn-kubernetes we can then configure
+ovn-kubernetes-using OCP clusters as follows:
+
+  1. We set `host-network-namespace=openshift-host-network` in the
+     ovn-kubernetes config
+
+  2. CNO creates an `openshift-host-network` namespace, with the label
+     `network.openshift.io/policy-group: host-network`
+
+And for openshift-sdn clusters:
+
+  1. CNO ensures that the `default` namespace has the label
+     `network.openshift.io/policy-group: host-network`
```

Comment:
IMHO, `cluster-ingress-operator` should also annotate `default` with  `network.openshift.io/policy-group: router`